internal/app/machined/pkg/runtime/v1alpha1/v1alpha1_sequencer_tasks.go

```diff
@@ -128,6 +129,57 @@ func SetupSystemDirectory(seq runtime.Sequence, data interface{}) (runtime.TaskE
 	}, "setupSystemDirectory"
 }
 
+// CreateSystemCgroups represents the CreateSystemCgroups task.
+//
+//nolint:gocyclo
+func CreateSystemCgroups(seq runtime.Sequence, data interface{}) (runtime.TaskExecutionFunc, string) {
+	return func(ctx context.Context, logger *log.Logger, r runtime.Runtime) (err error) {
+		// in container mode cgroups mode depends on cgroups provided by the container runtime
+		if r.State().Platform().Mode() != runtime.ModeContainer {
+			// assert that cgroupsv2 is being used when running not in container mode,
+			// as Talos sets up cgroupsv2 on its own
+			if cgroups.Mode() != cgroups.Unified {
+				return fmt.Errorf("cgroupsv2 should be used")
+			}
+		}
```

Comment:
inside container you can create cgroups, on host machine it looks like subdirectories of base container

```bash
/sys/fs/cgroup/pids/docker/c9dee09ae012c9705c5cc2f93981155d499950fee693f02a43bc38254eb1fd83/system/runtime/
```
Init process has permission to move pids between its own cgroups

Reply:
yep, exactly, but I want to assert that when Talos runs standalone (not inside a container), it created proper unified cgroupsv2 hierarchy. Just to make sure we don't have any regression.